src/extern.h

```diff
@@ -134,4 +152,21 @@ int fido_verify_sig_rs256(const fido_blob_t *, const rs256_pk_t *,
 int fido_verify_sig_eddsa(const fido_blob_t *, const eddsa_pk_t *,
     const fido_blob_t *);
 
+/* OS specific dev_info_manifest */
+int fido_dev_info_manifest_linux(fido_dev_info_t *, size_t, size_t *);
+int fido_dev_info_manifest_win(fido_dev_info_t *, size_t, size_t *);
+int fido_dev_info_manifest_osx(fido_dev_info_t *, size_t, size_t *);
+int fido_dev_info_manifest_openbsd(fido_dev_info_t *, size_t, size_t *);
+int hidapi_dev_info_manifest(fido_dev_info_t *, size_t, size_t *);
+
+/* hid i/o */
+void *fido_hid_open(const char *);
+void fido_hid_close(void *);
+int fido_hid_read(void *, unsigned char *, size_t, int);
+int fido_hid_write(void *, const unsigned char *, size_t);
+
+/* device manifest registration */
+typedef int (*dev_manifest_func_t)(fido_dev_info_t *dev_infos, size_t ilen,
+                                   size_t *olen);
+int fido_dev_register_manifest_func(const dev_manifest_func_t func);
```

Comment:
Some of the additions appear to be duplicates.

Reply:
No, I don't think these are defined elsewhere.